lightshow/postprocess/Broaden.py

```diff
@@ -0,0 +1,361 @@
+"""Using known XAS exprimental spectrum and simulation to optimize the broadening pararmeters for other simulations"""
+
+
+from scipy.stats import pearsonr as ps
+from scipy.interpolate import interp1d
+from scipy.stats import spearmanr
+from scipy import optimize
+from scipy.stats import norm
+from scipy.stats import cauchy
+from scipy.special import voigt_profile as voigt
+
+import numpy as np
+
+
+def gauss_broaden(x, xin, yin, sigma, shift=0.0):
+    """Gaussian broadening function
+
+    Parameters
+    ----------
+    x: Array
+        The output energy grid.
+    xin: Array
+        The input energy grid.
+    yin: Array
+        The input spectrum
+    sigma: float
+        Gaussian broadening sigma
+    shift: float
+        The energy shift
+
+    Returns
+    -------
+    Array
+        Broadened spectrum
+
+    """
+    x1, x2 = np.meshgrid(x, xin + shift)
+
+    return (
+        np.dot(norm.pdf(x1, x2, sigma).T, yin) / len(xin) * (xin[-1] - xin[0])
+    )
+
+
+def lorentz_broaden(x, xin, yin, gamma, shift=0.0):
+    """Lorentzian broadening function
+
+    Parameters
+    ----------
+    x: Array
+        The output energy grid.
+    xin: Array
+        The input energy grid.
+    yin: Array
+        The input spectrum
+    gamma:float
+        Lorentzian broadening gamma
+    shift:float
+        The energy shift
+
+    Returns
+    -------
+    Array
+        Broadened spectrum
+
+    """
+    x1, x2 = np.meshgrid(x, xin + shift)
+
+    return (
+        np.dot(cauchy.pdf(x1, x2, gamma).T, yin) / len(xin) * (xin[-1] - xin[0])
+    )
+
+
+def voigt_broaden(x, xin, yin, sigma, gamma, shift=0.0):
+    """Voigt broadening function
+
+    Parameters
+    ----------
+    x: Array
+        The output energy grid.
+    xin: Array
+        The input energy grid.
+    yin: Array
+        The input spectrum
+    sigma:float
+        Voigt broadening sigma
+    gamma:float
+        Voigt broadening gamma
+    shift:float
+        The energy shift
+
+    Returns
+    -------
+    Array
+        Broadened spectrum
+
+    """
+    x1, x2 = np.meshgrid(x, xin + shift)
+
+    return (
+        np.dot(voigt(x1 - x2, sigma, gamma / 2).T, yin)
+        / len(xin)
+        * (xin[-1] - xin[0])
+    )
+
+
+class Site_spectum:
```

Comment:
I'd use the correct class notation here. Should be `SiteSpectrum` not `Site_spectrum`.